### DISCUSSION TITLE
Exception when adding ShouldQueue Listener to Event

### DISCUSSION THREAD

**Discussion by NiroDeveloper** [spatie/laravel-backup#1860] [Q&A]
If we add a ShouldQueue Listener to a event like BackupWasSuccessful it will throw a exception.
This is because laravel tries to serialize the Event, but the object contains a BackupDestination and that a Filesystem that can not be serialized.

Listener example:
```
class MyListener implements ShouldQueue
{
    public function handle(): void
    {

    }

}
```

In the service provider:
`Event::listen(BackupWasSuccessful::class, MyListener::class);`

It will throw a exception like this:
`ERROR: Serialization of 'finfo' is not allowed {"exception":{"class":"Exception","file":"/var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Queue.php","line":160,"trace":"#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(160): serialize()`

I testet this with the local and sftp filesystem driver.

**Comment by freekmurze** [comment]
I'm open to a PR that will fix this.

**Comment by nocodelab** [comment]
Hello guys, anyone was able to solve this issue?

**Reply by NiroDeveloper** [reply]
A workaround is to create a syncronized listener that dispatches a ShouldQueue-Job then.
